components/Tabs/src/private/MDCItemBar.h

```diff
@@ -77,13 +77,21 @@
  selection changes.
  */
 @protocol MDCItemBarDelegate <NSObject>
+@optional
```

Comment:
Since this is an internal, private interface let's be safe and keep everything required. You could actually delete willSelectItem from this protocol and just use shouldSelectItem to implement the MDCTabBar interface I think.